services/shuttle-rama/README.md

```diff
@@ -0,0 +1,73 @@
+## Shuttle service integration for the Rama framework
+
+Learn more about rama at <https://ramaproxy.org/> and see more [Rama v0.2] examples
+at <https://github.com/plabayo/rama/tree/rama-0.2.0/examples>.
+
+[Rama]: https://github.com/plabayo/rama
+
+### Examples
+
+#### Application Service
+
+```rust,ignore
+use rama::service::service_fn;
+use std::convert::Infallible;
+
+async fn hello_world() -> Result<&'static str, Infallible> {
+    Ok("Hello, world!")
+}
+
+#[shuttle_runtime::main]
+async fn main() -> Result<impl shuttle_rama::ShuttleService, shuttle_rama::ShuttleError> {
+    Ok(shuttle_rama::RamaService::application(
+        service_fn(hello_world),
+    ))
+}
+```
+
+#### Transport Service
+
+```rust,ignore
+use rama::{net, service::service_fn};
+use std::convert::Infallible;
+use tokio::io::AsyncWriteExt;
+
+async fn hello_world<S>(mut stream: S) -> Result<(), Infallible>
+where
+    S: net::stream::Socket + net::stream::Stream + Unpin,
+{
+    println!(
+        "Incoming connection from: {}",
+        stream
+            .peer_addr()
+            .map(|a| a.to_string())
+            .unwrap_or_else(|_| "???".to_owned())
+    );
```

Comment:
One thing to note here is that this will always be the address of the Shuttle proxy, so it's likely not achieving what you want. The way to get the peer IP of the original client currently is to get it from a proxy header: https://docs.shuttle.dev/docs/deployment-environment#https-traffic. Should we alter the example here and in the examples repo?